src/RoomNotifs.ts

```diff
@@ -189,12 +189,21 @@ function findOverrideMuteRule(roomId: string): IPushRule | null {
     return null;
 }
 
-function isRuleForRoom(roomId: string, rule: IPushRule): boolean {
+export function isRuleRoomSpecific(rule: IPushRule): boolean {
```

Comment:
A room-specific rule is a different thing to what this function appears to be looking for (https://spec.matrix.org/v1.6/client-server-api/#push-rules).

I'd suggest finding a different name, but a comment would also be good, defining exactly what it matches.

The fact it only matches rules with a single condition is a bit obscure: why doesn't it include rules which have multiple conditions (of which one is room match)? I can easily see future developers deciding to change that, which seems like it would break `isRuleForRoom`, so it seems particularly important to call out in a comment.

Reply:
Yeah, fair call.
I've (hopefully) simplified this a bit, and added some comments